src/coprocessor/codec/data_type/vector.rs

```diff
@@ -662,16 +675,22 @@ impl_as_slice! { DateTime, as_date_time_slice }
 impl_as_slice! { Duration, as_duration_slice }
 impl_as_slice! { Json, as_json_slice }
 
-macro_rules! impl_push {
-    ($ty:tt, $name:ident) => {
+pub trait VectorValueExt<T: Evaluable> {
```

Comment:
Could you please add some comments to describe it?

Reply:
Done, PTAL again, thanks!